src/SignalR/common/Http.Connections/src/ConnectionEndpointRouteBuilderExtensions.cs

```diff
@@ -86,6 +80,19 @@ public static class ConnectionEndpointRouteBuilderExtensions
         /// <returns>An <see cref="IEndpointConventionBuilder"/> for endpoints associated with the connections.</returns>
         public static IEndpointConventionBuilder MapConnections(this IEndpointRouteBuilder endpoints, string pattern, HttpConnectionDispatcherOptions options, Action<IConnectionBuilder> configure)
         {
+            IList<IAuthorizeData> authorizationData = options.AuthorizationData;
+            // Copy the list if there are items, we're going to clear the list on the options
+            // so that EndPoints is the only one running Auth and we don't run it again in HttpConnectionDispatcher
+            if (options.AuthorizationData.Count > 0)
+            {
+                authorizationData = new List<IAuthorizeData>(options.AuthorizationData.Count);
+                for (var i = 0; i < options.AuthorizationData.Count; i++)
+                {
+                    authorizationData.Add(options.AuthorizationData[i]);
+                }
+                options.AuthorizationData.Clear();
```

Comment:
Isn't a little weird to mutate the options object? What if someone tries to reuse HttpConnectionDispatcherOptions? Is there an alternative way to ensure HttpConnectionDispatcher doesn't run auth again?

Reply:
Or like one of my other suggestions, disallow this entirely and get users to use the endpoint routing authorization extensions: https://github.com/aspnet/AspNetCore/blob/16a47948f80fede807fabe3c291d793590e8fd17/src/Security/Authorization/Policy/src/AuthorizationEndpointConventionBuilderExtensions.cs#L14

While `HttpConnectionDispatcherOptions.AuthorizeData` is old API, the endpoint routing based extension methods are new. It doesn't seem terrible to say this feature does not work when using endpoint routing. We could also use the Startup analyzer to guide users to not use this property.